samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -19,109 +19,119 @@
 import org.apache.samza.coordinator.stream.messages.SetContainerHostMapping;
-
+import org.apache.samza.metadatastore.MetadataStore;
+import org.apache.samza.metadatastore.MetadataStoreFactory;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.serializers.Serde;
+import org.apache.samza.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Locality Manager is used to persist and read the container-to-host
- * assignment information from the coordinator stream
+ * assignment information from the coordinator stream.
  * */
 public class LocalityManager {
-  private static final String CONTAINER_PREFIX = "SamzaContainer-";
   private static final Logger LOG = LoggerFactory.getLogger(LocalityManager.class);
 
-  private final CoordinatorStreamManager coordinatorStreamManager;
+  private final Config config;
+  private final Serde<String> keySerde;
+  private final Serde<String> valueSerde;
+  private final MetadataStore metadataStore;
   private final TaskAssignmentManager taskAssignmentManager;
 
-  private Map<String, Map<String, String>> containerToHostMapping = new HashMap<>();
+  /**
+   * Builds the LocalityManager based upon {@link Config} and {@link MetricsRegistry}.
+   * Uses {@link CoordinatorStreamKeySerde} and {@link CoordinatorStreamValueSerde} to
+   * serialize messages before reading/writing into coordinator stream.
+   *
+   * @param config the configuration required for setting up metadata store.
+   * @param metricsRegistry the registry for reporting metrics.
+   */
+  public LocalityManager(Config config, MetricsRegistry metricsRegistry) {
+    this(config, metricsRegistry, new CoordinatorStreamKeySerde(SetContainerHostMapping.TYPE),
+         new CoordinatorStreamValueSerde(config, SetContainerHostMapping.TYPE));
+  }
 
   /**
-   * Constructor that creates a read-write or write-only locality manager.
+   * Builds the LocalityManager based upon {@link Config} and {@link MetricsRegistry}.
+   * Uses keySerde, valueSerde to serialize/deserialize (key, value) pairs before reading/writing
+   * into {@link MetadataStore}.
    *
-   * @param coordinatorStreamManager Coordinator stream manager.
+   * Key and value serializer are different for yarn (uses CoordinatorStreamMessage) and standalone (native ObjectOutputStream for serialization) modes.
+   * @param config the configuration required for setting up metadata store.
+   * @param metricsRegistry the registry for reporting metrics.
+   * @param keySerde the key serializer.
+   * @param valueSerde the value serializer.
    */
-  public LocalityManager(CoordinatorStreamManager coordinatorStreamManager) {
-    this.coordinatorStreamManager = coordinatorStreamManager;
-    this.taskAssignmentManager = new TaskAssignmentManager(coordinatorStreamManager);
+  public LocalityManager(Config config, MetricsRegistry metricsRegistry, Serde<String> keySerde, Serde<String> valueSerde) {
```

Comment:
Is this constructor used anywhere (other than tests)? If not, maybe make protected/package private.

Reply:
Done.